protocol-units/mempool/util/src/lib.rs

```diff
@@ -122,7 +122,14 @@ impl Ord for MempoolTransaction {
 			Ordering::Equal => {}
 			non_equal => return non_equal,
 		}
-		// If slot_seconds are equal, then compare by transaction
+
+		// If slot_seconds are equal, then compare by sequence number
+		match self.transaction.sequence_number.cmp(&other.transaction.sequence_number) {
+			Ordering::Equal => {}
+			non_equal => return non_equal,
+		}
+
+		// If sequence number is equal, then compare by transaction on the whole
 		self.transaction.cmp(&other.transaction)
```

Comment:
This theoretically can compare the sequence numbers again, but in a very corner case. This last resort comparison is OK as a backstop to achieve a stable order without accounting for any new field we could add to the `Transaction` struct in the future.

But you could achieve the same effect implicitly using just this derived `PartialOrd` impl, if you put the `sequence_number` field first in the struct definition.

Reply:
Will change.

Reply:
To think of it, it might be better to not violate layering here and move the sequence number comparison into (explicitly defined and commented) `impl PartialOrd for Transaction`.

Reply:
> Instead of `Ord`?

Oh yes, both. Normally you just proxy `Ord` to `PartialOrd` which in fact implements a total order.
Or just derive both, but put the `sequence_number` member first in the struct.